secboot/secboot.go

```diff
@@ -106,3 +106,37 @@ type UnlockVolumeUsingSealedKeyOptions struct {
 	// will be attempted if activation with the sealed key failed.
 	AllowRecoveryKey bool
 }
+
+// UnlockMethod is the method that was used to unlock a volume.
+type UnlockMethod int
+
+const (
+	// NotUnlocked indicates that the device was either not unlocked or is not
+	// an encrypted device.
+	NotUnlocked UnlockMethod = iota
+	// UnlockedWithSealedKey indicates that the device was unlocked with the
+	// provided sealed key object.
+	UnlockedWithSealedKey
+	// UnlockedWithRecoveryKey indicates that the device was unlocked by the
+	// user providing the recovery key at the prompt.
+	UnlockedWithRecoveryKey
+	// UnlockedWithUnsealedKey indicates that the device was unlocked with the
+	// provided key object that is not sealed.
+	UnlockedWithUnsealedKey
```

Comment:
Got confused by this a little. A sealed key gets unsealed, so basically becomes an 'unsealed key' :smile:. Anyways, maybe we should drop it for now until there is a need for this status?